src/components/Header/Header.tsx

```diff
@@ -14,13 +14,33 @@ import {
 
 import { Button } from "~/components/ui/button";
 
+import { Switch } from "~/components/ui/switch";
+
+import { Sun, Moon } from "lucide-react";
+
+import { useDispatch, useSelector } from 'react-redux';
+import { setIsDay, AppDispatch, RootState } from '../../rtk';
+
+import { MouseEvent } from 'react';
+
 const pages = [
   { id: 1, pathname: "Home" },
   { id: 2, pathname: "Login" },
   { id: 3, pathname: "Editor" },
 ] as const;
 
 export const Header = () => {
+
+  const clickHandler = () => document.body.classList.toggle('dark');
+
+  const dispatch = useDispatch<AppDispatch>();
+  const { isDay } = useSelector((state: RootState) => state.data);
```

Comment:
please create properly typed versions of useSelector and useDispatch in one file and then use it throughout the project.